src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Setup.csproj

```diff
@@ -52,11 +52,55 @@
       <Output TaskParameter="CopiedFiles" ItemName="FileWrites"/>
     </Copy>
   </Target>
-  
-  <Target Name="_GenerateSwrFile" 
+
+  <!--
+  Splits the $(XlfLanguages) property into an item group.  E.g., this:
+
+    <PropertyGroup>
+      <XlfLanguages>cs;de;...</XlfLanguages>
+    </PropertyGroup>
+
+  turns into this:
+
+    <ItemGroup>
+      <_XlfLanguages Include="cs" />
+      <_XlfLanguages Include="de" />
+      ...
+    </ItemGroup>
+
+   -->
+  <Target Name="_GenerateLanguageList">
+    <CreateItem Include="$(XlfLanguages)">
+      <Output TaskParameter="Include" ItemName="_XlfLanguages" />
+    </CreateItem>
+  </Target>
+
+  <!-- For each localization language, include the appropriate satellite assemblies. -->
+  <Target Name="_GenerateLocSwrFileContents"
+          DependsOnTargets="_GenerateLanguageList"
+          Outputs="%(_XlfLanguages.Identity)">
```

Comment:
This is implicitly targeting on a file which won't actually exist. Is this just to force MSBuild batching?

Reply:
Yes, that's correct.

Reply:
Do we need target batching here? I think the following should work and it's simpler (no need for extra targets):

```
<ItemGroup>
  <_XlfLanguages Include="$(XlfLangauges)"/>

   <_SwrLinesPerLanguage Include="@(_XlfLanguages)">
     <Lines><![CDATA[... %(_XlfLanguages.Identity) ...]]></Lines>
   </_SwrLinesPerLanguage>
</ItemGroup>
```